remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/mapping/reference/FieldReferenceProvider.java

```diff
@@ -0,0 +1,122 @@
+package org.infinispan.query.remote.impl.mapping.reference;
+
+import org.hibernate.search.engine.backend.document.IndexFieldReference;
+import org.hibernate.search.engine.backend.document.model.dsl.IndexSchemaElement;
+import org.hibernate.search.engine.backend.document.model.dsl.IndexSchemaFieldOptionsStep;
+import org.hibernate.search.engine.backend.types.Aggregable;
+import org.hibernate.search.engine.backend.types.Projectable;
+import org.hibernate.search.engine.backend.types.Searchable;
+import org.hibernate.search.engine.backend.types.Sortable;
+import org.hibernate.search.engine.backend.types.dsl.IndexFieldTypeFactory;
+import org.hibernate.search.engine.backend.types.dsl.IndexFieldTypeFinalStep;
+import org.hibernate.search.engine.backend.types.dsl.StandardIndexFieldTypeOptionsStep;
+import org.hibernate.search.engine.backend.types.dsl.StringIndexFieldTypeOptionsStep;
+import org.infinispan.commons.logging.LogFactory;
+import org.infinispan.protostream.descriptors.FieldDescriptor;
+import org.infinispan.protostream.descriptors.Type;
+import org.infinispan.query.remote.impl.indexing.FieldMapping;
+import org.infinispan.query.remote.impl.logging.Log;
+
+public class FieldReferenceProvider {
+
+   private static final Log log = LogFactory.getLog(FieldReferenceProvider.class, Log.class);
+
+   private final String name;
+   private final Type type;
+
+   private final Searchable searchable;
+   private final Projectable projectable;
+   private final Aggregable aggregable;
+   private final Sortable sortable;
+
+   private final String analyzer;
+   private final Object indexNullAs;
+   private final boolean repeated;
+
+   public FieldReferenceProvider(FieldDescriptor fieldDescriptor, FieldMapping fieldMapping) {
+      // the property name and type are taken from the model
+      name = fieldDescriptor.getName();
+      type = fieldDescriptor.getType();
+
+      analyzer = (fieldMapping.analyze()) ? fieldMapping.analyzer() : null;
+
+      // the rest from the mapping
+      searchable = (fieldMapping.index()) ? Searchable.YES : Searchable.NO;
+      sortable = (fieldMapping.sortable()) ? Sortable.YES : Sortable.NO;
+
+      projectable = (fieldMapping.store()) ? Projectable.YES : Projectable.NO;
+      // TODO verify when it is possible to make the field aggregable
```

Comment:
I'd say never, since Infinispan aggregations are implemented without relying on Hibernate Search?

Reply:
you're right (as usual :) )... yes the aggregation are derived on top of simple projections queries by ISPN... so it seems that ISPN doesn't need the Search 6's aggregation. I'll remove the comment.